src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -169,6 +173,53 @@ def pex_requirements_for_default_lockfile(cls):
             resolve_name=cls.options_scope,
         )
 
+    @classproperty
+    def help_extended(cls) -> str:
+        base_help = cls.help if isinstance(cls.help, str) else cls.help()
+        if cls.default_lockfile_resource is None:
+            return base_help
+
+        lockfile = cls.pex_requirements_for_default_lockfile()
+        parts = urlparse(lockfile.url)
+        # urlparse retains the leading / in URLs with a netloc.
+        lockfile_path = parts.path[1:] if parts.path.startswith("/") else parts.path
+        if parts.scheme in {"", "file"}:
+            with open(lockfile_path, "rb") as fp:
+                lock_bytes = fp.read()
+        elif parts.scheme == "resource":
+            _fc = FileContent(
+                lockfile_path,
+                # The "netloc" in our made-up "resource://" scheme is the package.
+                importlib.resources.read_binary(parts.netloc, lockfile_path),
+            )
+            lockfile_path, lock_bytes = (_fc.path, _fc.content)
+        else:
+            raise ValueError(
+                f"Unsupported scheme {parts.scheme} for lockfile URL: {lockfile.url} "
+                f"(origin: {lockfile.url_description_of_origin})"
+            )
+
+        stripped_lock_bytes = strip_comments_from_pex_json_lockfile(lock_bytes)
+        lockfile_contents = json.loads(stripped_lock_bytes)
+        package_version = next(
+            (
+                requirement["version"]
+                for resolve in lockfile_contents["locked_resolves"]
+                for requirement in resolve["locked_requirements"]
+                if requirement["project_name"] == cls.options_scope
```

Comment:
I note that there's some tools for which the options scope isn't the same as the underlying package e.g. `HelmKubeParserSubsystem` has `options_scope = "helm-k8s-parser"` but uses the `hikaru` package.

Just brainstorming an idea:

1. extract the project name from the first element of `default_requirements` (and document that the first element should be the "main" one) using `pants.util.pip_requirement.PipRequirement`
2. check that against the `project_name` instead (we may need to call `packaging.utils.canonicalize_name` first; there's a few other places in the codebase that import `from packaging.utils import canonicalize_name as canonicalize_project_name`)
3. change this search to _require_ a `package_version` (i.e. if we make that change, I think there's no circumstances where we wouldn't have it)
4. update the rendered text to use the project name too, not `cls.options_scope`

(The change 3 also seems good for ensuring this feature continues to work: if we break the code somehow, it'll fail fast, rather than silently failing to render.)

Reply:
Makes sense! For this:

> 1. extract the project name from the first element of default_requirements (and document that the first element should be the "main" one) using `pants.util.pip_requirement.PipRequirement`

This is entirely by convention, right? There's no other way to enforce this to be true (as far as I know).

I pushed up a commit containing all the other changes in this list.

Reply:
> This is entirely by convention, right? There's no other way to enforce this to be true (as far as I know).

Yeah, no way to force it that I know of either.